### DISCUSSION TITLE
How to set a time?

### DISCUSSION THREAD

**Discussion by indyjonesnl** [spatie/schema-org#202] [Q&A] [answered]
```
$lodgingBusiness = Schema::lodgingBusiness()
  ->openingHours(
    Schema::openingHoursSpecification()
      ->dayOfWeek([Schema::dayOfWeek()::Monday])
      ->opens(new DateTime('09:00:00'))
      ->closes(new DateTime('17:00:00'))
  )
  ->checkinTime(new DateTime('14:00:00'))
  ->checkoutTime(new DateTime('11:00:00'))
```

This results in the current **date + time** being included in the output `"opens":"2024-01-10T09:00:00+00:00","closes":"2024-01-10T17:00:00+00:00"` and `..."checkinTime":"2024-01-10T14:00:00+00:00","checkoutTime":"2024-01-10T11:00:00+00:00"...`

While the Schema should contain only the **time**, formatted as `14:30:00+08:00`.

Can someone point me in the right direction?

**Comment by Gummibeer** [comment, marked as answer]
That's a problem right now - `Time`, `Date` and `DateTime` schema.org types are all assigned the `DateTimeInterface` php type.
https://github.com/spatie/schema-org/blob/111488e5110da30cc96b775b480bc29b28164883/generator/Parser/Tasks/ParseProperty.php#L51-L54
And the `DateTimeInterface` php type is always formatted as atom string.
https://github.com/spatie/schema-org/blob/111488e5110da30cc96b775b480bc29b28164883/generator/templates/static/BaseType.php#L124-L126

It would need some additional data passing from the spec parser to the payload serializer or some additional interfaces - which would have to be custom as php doesn't have specific date or time only interfaces.
The current workaround would be to call the underlying `setProperty()` method directly with a simple string.
